pkg/types/vsphere/validation/platform.go

```diff
@@ -38,6 +40,24 @@ func ValidatePlatform(p *vsphere.Platform, fldPath *field.Path) field.ErrorList
 	// If all VIPs are empty, skip IP validation.  All VIPs are required to be defined together.
 	if strings.Join([]string{p.APIVIP, p.IngressVIP}, "") != "" {
 		allErrs = append(allErrs, validateVIPs(p, fldPath)...)
+		if network != nil {
```

Comment:
Networking must never be nil. So rather than ignore a nil, at least add an error for it.